src/main/java/io/supertokens/dashboard/Dashboard.java

```diff
@@ -0,0 +1,190 @@
+        DashboardSQLStorage storage = StorageLayer.getDashboardStorage(main);
+        try {
+            storage.startTransaction(transaction -> {
+                if (newEmail != null) {
+                    try {
+                        storage.updateDashboardUsersEmailWithUserId_Transaction(transaction, userId, newEmail);
+                    } catch (DuplicateEmailException e) {
+                        throw new StorageTransactionLogicException(e);
+                    }
+                }
+
+                if (newPassword != null) {
+                    String hashedPassword = PasswordHashing.getInstance(main).createHashWithSalt(newPassword);
+                    storage.updateDashboardUsersPasswordWithUserId_Transaction(transaction, userId, hashedPassword);
+                }
+                storage.commitTransaction(transaction);
+                return null;
+            });
+        } catch (StorageTransactionLogicException e) {
+            if (e.actualException instanceof DuplicateEmailException) {
+                throw (DuplicateEmailException) e.actualException;
+            }
+            throw e;
+        }
+    }
+
+    public static boolean isFeatureFlagEnabledOrUserCountIsUnderThreshold(Main main) throws StorageQueryException {
+
+        if (!isDashboardFeatureFlagEnabled(main)) {
+            // retrieve current dashboard users
+            DashboardUser[] users = StorageLayer.getDashboardStorage(main).getAllDashboardUsers();
+            // check if current dashboard users count is under the threshold
+            return users.length <= MAX_NUMBER_OF_FREE_DASHBOARD_USERS;
+        }
+        return true;
+    }
+
+    private static boolean isDashboardFeatureFlagEnabled(Main main) throws StorageQueryException {
+        return Arrays.stream(FeatureFlag.getInstance(main).getEnabledFeatures())
+                .anyMatch(t -> t == EE_FEATURES.DASHBOARD_LOGIN);
+    }
+
+    private static String createSessionForDashboardUser(Main main, DashboardUser user) throws StorageQueryException {
+        String sessionId = UUID.randomUUID().toString();
+        long timeCreated = System.currentTimeMillis();
+        StorageLayer.getDashboardStorage(main).createNewDashboardUserSession(user.userId, sessionId, timeCreated);
+        return sessionId;
+    }
+
+    public static boolean isValidEmail(String email) {
+        // We use the same regex as the backend SDK
+        // https://github.com/supertokens/supertokens-node/blob/master/lib/ts/recipe/emailpassword/utils.ts#L250
+        String regexPatternForEmail = "((^<>()[].,;:@]+(.^<>()[].,;:@]+)*)|(.+))@(([[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}])|(([a-zA-Z-0-9]+.)+[a-zA-Z]{2,}))$";
+        return patternMatcher(email, regexPatternForEmail);
+    }
+
+    public static boolean isStrongPassword(String password) {
+
+        String regexPatternForPassword = "(?=.*[A-Za-z])(?=.*[0-9]).{8,100}";
```

Comment:
Instead of using a regex for this, it might be better to use the same if else conditions that the default validator in node uses